### PR TITLE
Update README file

### PR DESCRIPTION
**Pull request by lyrm**
# Changes
- Add missing data structures in the README,
- Move examples into other files for readability.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,455 +81,113 @@ opam switch list-available
...
-  let q = create () in
-  let domainA = Domain.spawn (fun () -> work_par 1 barrier q) in
-  let domainB = Domain.spawn (fun () -> work_par 2 barrier q) in
-  Domain.join domainA;
-  Domain.join domainB;
-  (List.length !q, !q)
-```
+- **Module**: [Bounded_stack](https://ocaml-multicore.github.io/saturn/saturn/Bounded_stack/index.html)
+- **Description**: A stack based on the Treiber stack algorithm, with a limited capacity and a `length` function.
+- **Recommendation**: Adding a capacity introduces a general overhead to the operations. It is recommended to use the unbounded stack if neither the capacity nor the `length` function is needed.
```
Comment (art-w) [inline README.md:95]: Perhaps explain the positive side of having a max capacity?

File: README.md [hunk 2]
```diff
@@ -89,455 +81,113 @@ opam switch list-available
...
-    | Some elt -> pop_all (elt :: acc)
-  in
-  let res = pop_all [] in
-  (List.length res, res)
-```
+### Lock-free Skip List
 
-Running it results in the expected result:
+- **Module**: [Skiplist](https://ocaml-multicore.github.io/saturn/saturn/Skiplist/index.html)
+- **Description**: A skiplist is a probabilistic data structure that has an average logarithmic complexity for search and insertion operations.
```
Comment (art-w) [inline README.md:134]: Perhaps mention that it's an ordered collection? (similar to a Map)

File: README.md [hunk 3]
```diff
@@ -89,455 +81,113 @@ opam switch list-available
...
 
-```ocaml
-# test_saturn ();;
-- : int * int list =
-(20, [2; 2; 1; 2; 2; 2; 2; 2; 2; 1; 2; 1; 2; 1; 1; 1; 1; 1; 1; 1])
-```
+### Lock-free Hash Table
 
-### A note about races in OCaml
+- **Module**: [Hstbl](https://ocaml-multicore.github.io/saturn/saturn/htbl/index.html)
```
Comment (art-w) [inline README.md:140]: typo: Htbl

**Review by art-w — CHANGES_REQUESTED** [review]
I much prefer the new layout instead of the previous table! The links to the documentation appear to be broken however

File: README.md (hunk 1 above)
Comment (lyrm) [reply README.md:95]: It's memory bounded. Are you thinking of anything else ?

Comment (art-w) [reply README.md:95]: No that's it!

**Pull request merged**